### DISCUSSION TITLE
Receive IDocs with PyRFC server

### DISCUSSION THREAD

**Discussion by trashutz** [SAP/PyRFC#309] [Q&A] [answered]
I saw the example `clientIDoc.py` in which IDoc data is send to a server.

My question is: Can I start a server with PyRFC, which receives IDoc data? And how could I implement that?

Can I just bind the `IDOC_INBOUND_ASYNCHRONOUS` RFC call to custom function that parses the IDoc data like that:
````python
server.add_function("IDOC_INBOUND_ASYNCHRONOUS", my_idoc_parsing_function)
````

Or is there another way?

**Comment by bsrdjan** [comment, marked as answer]
Hello @trashutz,

Python server supports transactional RFC protocol required for sending IDoc from ABAP to SAP system. Assuming the ABAP sender system is properly configured, the connection to Python server can be established and Python server can receive IDoc data. 

As @Ulrich-Schmidt explained in https://github.com/SAP/node-rfc/discussions/279#discussioncomment-4905375, in response to similar question, that is only "half the battle", because parsing IDoc data is more challenging part here. SAP IDoc library for solving that problem is available for Java only and for SAP Java RFC connector JCo. One usage example is given in SAP Help [Example Program IDoc Server](https://help.sap.com/saphelp_ewm900/helpdata/en/41/c78586b4f349dc90d522a75cb1a5bd/frameset.htm) but counterpart in Python, or another language is not available because IDoc parsing library is provided in Java only. You could also consider SAP Business Connector as recommended in linked comment.

**Reply by trashutz** [reply]
@bsrdjan Thank you! I think I'll take another look at the JCO and Java IDoc library.

**Reply by bsrdjan** [reply]
Check also: https://github.com/SAP/PyRFC/discussions/340#discussioncomment-7326815
